src/components/DeploymentComboboxOption.vue

```diff
@@ -1,48 +1,37 @@
 <template>
   <div ref="element" class="deployment-combobox-option">
-    <span v-if="visible && flowResponse" class="deployment-combobox-option--flow-name">
-      {{ flowResponse.name }}
-      <p-icon icon="ChevronRightIcon" size="small" class="deployment-combobox-option--chevron" />
-    </span>
+    <template v-if="flow">
+      <span class="deployment-combobox-option__flow-name">
+        {{ flow.name }}
+      </span>
+      <p-icon icon="ChevronRightIcon" size="small" class="deployment-combobox-option__chevron" />
+    </template>
     {{ deploymentName }}
   </div>
 </template>
 
 <script lang="ts" setup>
-  import { useVisibilityObserver, useSubscriptionWithDependencies } from '@prefecthq/vue-compositions'
-  import { computed, ref } from 'vue'
-  import { useWorkspaceApi } from '@/compositions'
+  import { ref } from 'vue'
+  import { useFlow } from '@/compositions'
 
   const props = defineProps<{
     flowId: string,
     deploymentName?: string,
   }>()
 
-  const api = useWorkspaceApi()
   const element = ref<HTMLDivElement>()
-  const { visible } = useVisibilityObserver(element, { disconnectWhenVisible: true })
-
```

Comment:
I don't think we need this element ref now.

Reply:
true! removed